app/src/main/java/org/eyeseetea/malariacare/LoginActivity.java

```diff
@@ -61,24 +77,41 @@
      */
     private String password;
 
-    @Override
+   @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        //Populate server with the current value
+
+       if (User.getLoggedUser() != null && !ProgressActivity.PULL_CANCEL ) {
+            startActivity(new Intent(LoginActivity.this,
+                   ((Dhis2Application) getApplication()).getMainActivity()));
+
+           finish();
+
+
+       }
+       ProgressActivity.PULL_CANCEL =false;
+           //Populate server with the current value
         EditText serverText = (EditText) findViewById(org.hisp.dhis.android.sdk.R.id.server_url);
         serverText.setText(ServerAPIController.getServerUrl());
         //Readonly
         //serverText.setEnabled(false);
 
-        //Username, Password blanks to force real login
-        EditText usernameEditText = (EditText) findViewById(R.id.username);
-        usernameEditText.setText("");
-        EditText passwordEditText = (EditText) findViewById(R.id.password);
-        passwordEditText.setText("");
+       SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+       EditText passwordEditText = (EditText) findViewById(R.id.password);
+       EditText usernameEditText = (EditText) findViewById(R.id.username);
+       if (sharedPreferences.getString("","").isEmpty()) {
```

Comment:
@Strabagkantza why this if? I mean, you're not asking for as string to the preferences, but putting "" where the key should be. That's always going to answer "" right? Is there any situation in which this would give a different answer?